### PR TITLE
Update to bytes 1.0

### PR DESCRIPTION
**Pull request by nickelc**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -155,15 +155,15 @@ impl<'t> DoRead<'t> {
...
             // We first have to initialize the data or otherwise casting to a byte slice
             // below is UB. See also code of std::io::copy(), tokio::AsyncRead::poll_read_buf()
             // and others.
             //
             // Read::read() might read uninitialized data otherwise, and generally creating
             // references to uninitialized data is UB.
-            for x in data.iter_mut() {
-                *x.as_mut_ptr() = 0;
+            for i in 0..data.len() {
+                data.write_byte(i, 0);
```
Comment (sdroege) [inline src/lib.rs:166]: Or maybe instead of that we can do something with `bytes::buf::UninitSlice` here without having to initialize everything first?

**Review by sdroege — APPROVED** [review]
Seems good to me otherwise, that can also be done in a separate PR IMHO.

File: Cargo.toml [hunk 2]
```diff
@@ -12,4 +12,4 @@ documentation = "https://docs.rs/input_buffer/0.3.1"
 version = "0.3.1"
```
Comment (sdroege) [inline Cargo.toml]: AFAIU this changes the API so the next release should be 0.4.0

**Comment by gakonst** [comment]
@application-developer-DA is there anything blocking this PR from being merged? There's a few upstream crates depending on it so would be great if we could get it in / release. Thank you!

**Review by daniel-abramov — APPROVED** [review]

**Comment by gakonst** [comment]
Thank you @application-developer-DA. Do you also mind releasing 0.4.0 to crates?

**Comment by daniel-abramov** [comment]
Sure.

@agalakhov , could you please publish it? (or give me permissions to do so, I'm not an owner of the crate ;)

**Comment by gakonst** [comment]
Bump @agalakhov it'd be great if you could do the release / give DA perms! Thank you again.

**Pull request merged**